From: Christian Barcenas <christian@cbarcenas.com>
Date: Tue, 10 Mar 2020 23:40:49 -0700
Subject: Port the build to Python 3

The build relies on Python to generate a linker script and to
populate the Makefile's VERSION variable.

Since Python 2 is EOL in bullseye, we need to update some
shebang lines and change one instance of 'print' to 'print(...)'.

---
 configure.self           | 2 +-
 lds-gen.py               | 2 +-
 packaging/get_version.py | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.self b/configure.self
index f6948c3..72db939 100644
--- a/configure.self
+++ b/configure.self
@@ -260,7 +260,7 @@ void foo (void) {
 
     # The linker-script generator (lds-gen.py) requires python
     if [[ $WITH_LDS == y ]]; then
-        if ! mkl_command_check python "HAVE_PYTHON" "disable" "python -V"; then
+        if ! mkl_command_check python3 "HAVE_PYTHON" "disable" "python3 -V"; then
             mkl_err "disabling linker-script since python is not available"
             mkl_mkvar_set WITH_LDS WITH_LDS "n"
         fi
diff --git a/lds-gen.py b/lds-gen.py
index b3ad9fb..a32070f 100755
--- a/lds-gen.py
+++ b/lds-gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # librdkafka - Apache Kafka C library
 #
diff --git a/packaging/get_version.py b/packaging/get_version.py
index 3d98d21..fad1d97 100755
--- a/packaging/get_version.py
+++ b/packaging/get_version.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import sys
 
@@ -18,4 +18,4 @@ minor = int(version[4:6], 16)
 patch = int(version[6:8], 16)
 version = '.'.join(str(item) for item in (major, minor, patch))
 
-print version
+print(version)
